### PR TITLE
create User model, migration files and tests

### PR DESCRIPTION
**Pull request by AshkenYariv**
Creating a user model, a static method for creating users.
Made migration files for user.
Wrote tests for user creation.
Deleted joboard application from the first demo since
it is not relevant anymore.

### CODE REVIEW

**Comment by yuvalk** [comment]
plz separate the removal of the board content from the adding of the user model.
(so you should have 2 commits, one that remove stuff and another one that add new stuff)

**Comment by yuvalk** [comment]
also, try to be as extensive as possible with tests, cover all the options. try to think about edge cases

**Comment by shayakrach** [comment]
@yuvalk  We split up the PR.

**Review by shayakrach — APPROVED** [review]

**Review by shayakrach — APPROVED** [review]

**Review by assafg95 — APPROVED** [review]

**Review by edenmaor — APPROVED** [review]

**Review by ormergi — COMMENTED** [review]
Regarding Commit "Deletion of jobboard app", it can be extracted to its own PR, as its not directly related to adding models.
Also keeping PR's small reduce code-review cycles and keep your reviewers focused.

File: side_hustle/settings.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'jobboard.apps.JobboardConfig',
```
Comment (ormergi) [inline side_hustle/settings.py:40]: This changes is related to removing jobboard app commit, isnt it?

File: side_hustle/urls.py [hunk 2]
```diff
@@ -15,9 +15,7 @@
 """
 from django.contrib import admin
 from django.urls import path
-from jobboard import views
 
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path('', views.board, name='board')
```
Comment (ormergi) [inline side_hustle/urls.py:22]: Same goes for these, they are related to the commit before this one, right?

**Review by ormergi — COMMENTED** [review]
Regarding "Create user model, migration files and tests" commit
The commit title should not describe adding tests or migration files, these are implementation details that could be understand by the actual changes this commit beings.
It can be rephrased to something like _'Add 'User' data model"_.
Same goes for the the commit message body, the "what" is described byt the actual changes this commit brings (add/remove/change files).

**Review by ormergi — COMMENTED** [review]
Regarding "Creation jobs model, migration files and tests" commit, see my comment about the previous commit https://github.com/redhat-beyond/Side-Hustle/pull/21#pullrequestreview-817147622.

File: side_hustle/settings.py (hunk 1 above)
Comment (shayakrach) [reply side_hustle/settings.py:40]: Right.

**Comment by shayakrach** [comment]
@ormergi 
We tried to keep the first commit separate and continue with another branch not dependent on it, but the jobborad app caused some errors. What should we do in that situation? And can we fix the current situation?

**Review by ormergi — COMMENTED** [review]
Regarding "Register users and jobs model to admin" commit,
correct me if I am wrong, but it looks like this commit has more to it other than what the message describes.
Those changes should be moved to new commit(s) when needed

**Comment by ormergi** [comment]
> @ormergi We tried to keep the first commit separate and continue with another branch not dependent on it, but the jobborad app caused some errors. What should we do in that situation? And can we fix the current situation?

There are few changes on the **second** commit that related to the **first** one, maybe its related.
Move the related changes to the first commit and try again.

File: side_hustle/settings.py (hunk 1 above)
Comment (ormergi) [reply side_hustle/settings.py:40]: Please move it to the first commit

**Comment by shayakrach** [comment]
@ormergi We understand **what** is the problem but we need help with the git technicalities.

**Comment by AshkenYariv** [comment]
Hello @ormergi these comit have been up on GItHub since sunday and we made correction according to what Edward and Yuval told us, these are too many changes that should have came up at an earlier time. What do you suggest we do taking into account that tomorrow we need to show Demo#2?

**Comment by shayakrach** [comment]
@ormergi We opened new PR #24.

**Comment by EdDev** [comment]
Please rebase over the updated master, this PR has conflicts now.

**Comment by AshkenYariv** [comment]
This PR was split up as asked, the reviewing should start at PR create user model

**Comment by ormergi** [comment]
@AshkenYariv its good that you explain why you close the PR, it will be awesome if you also paste the like for the new one.
Also, my code review comments may help you in following PR's as well.

EDIT: Cant find the following PR, if its not published yet please mention it.

**Comment by shayakrach** [comment]
Hi @ormergi, 
We forgot to paste it.
The following PR is #25.
